image_transport/include/image_transport/subscriber_plugin.hpp

```diff
@@ -146,7 +150,8 @@ class IMAGE_TRANSPORT_PUBLIC SubscriberPlugin
   virtual void subscribeImpl(
     rclcpp::Node * node, const std::string & base_topic,
     const Callback & callback,
-    rmw_qos_profile_t custom_qos = rmw_qos_profile_default) = 0;
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
+    SubscriptionOptionsWithVoidAllocator options = SubscriptionOptionsWithVoidAllocator()) = 0;
```

Comment:
isn't this breaking API?
IIUC, `SubscriberPlugin` (or `SimpleSubscriberPlugin`) are actually extended externally

Reply:
Good catch. I think we should try to make this change in an API compatible way, if possible.

Reply:
I would use a default method that ignores the provided options, similar to what I suggested [here](https://github.com/ros2/message_filters/pull/53#discussion_r626682459).

It's not awesome, but there aren't many other options.
The default method could just throw an error instead of warn, so the plugin only continues working where the new overolad isn't used. But that isn't great either ...

Reply:
You're right. Should be fixed now.